zhaquirks/xiaomi/aqara/motion_agl1.py

```diff
@@ -0,0 +1,179 @@
+
+RESET_NO_PRESENCE_STATUS_WRITE_VALUE = 1
+RESTART_DEVICE_WRITE_VALUE = 0
+
+
+class AqaraMotionSensitivity(types.enum8):
+    """Aqara motion sensitivity."""
+
+    Low = 0x01
+    Medium = 0x02
+    High = 0x03
+
+
+class AqaraMotion(types.enum8):
+    """Aqara motion."""
+
+    Unknown_0 = 0x00
+    Unknown_1 = 0x01
+    Idle = 0x02
+    Moving = 0x03
+    Still = 0x04
+
+
+class OppleCluster(XiaomiAqaraE1Cluster):
+    """Aqara manufacturer cluster for the FP1E presence sensor."""
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Manufacturer specific attributes."""
+
+        approach_distance = ZCLAttributeDef(
+            id=APPROACH_DISTANCE_ATTR_ID,
+            type=types.uint32_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        motion = ZCLAttributeDef(
+            id=MOTION_ATTR_ID,
+            type=types.uint8_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        motion_distance = ZCLAttributeDef(
+            id=MOTION_DISTANCE_ATTR_ID,
+            type=types.uint32_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        motion_sensitivity = ZCLAttributeDef(
+            id=MOTION_SENSITIVITY_ATTR_ID,
+            type=types.uint8_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        occupancy = ZCLAttributeDef(
+            id=OCCUPANCY_ATTR_ID,
+            type=types.uint8_t,
+            access="rp",
+            is_manufacturer_specific=True,
+        )
+
+        reset_no_presence_status = ZCLAttributeDef(
+            id=RESET_NO_PRESENCE_STATUS_ATTR_ID,
+            type=types.uint8_t,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+        restart_device = ZCLAttributeDef(
+            id=RESTART_DEVICE_ATTR_ID,
+            type=types.Bool,
+            access="w",
+            is_manufacturer_specific=True,
+        )
+
+    def _update_attribute(self, attrid: int, value: Any) -> None:
+        super()._update_attribute(attrid, value)
+        if attrid == OCCUPANCY_ATTR_ID:
+            self.endpoint.occupancy.update_attribute(
+                OccupancySensing.AttributeDefs.occupancy.id,
+                OccupancySensing.Occupancy(value),
+            )
+        elif attrid == MOTION_ATTR_ID:
+            self.endpoint.ias_zone.update_attribute(
+                IasZone.AttributeDefs.zone_status.id,
+                IasZone.ZoneStatus(
+                    IasZone.ZoneStatus.Alarm_1 if value == AqaraMotion.Moving else 0
+                ),
+            )
+
+
+(
+    QuirkBuilder("aqara", "lumi.sensor_occupy.agl1")
+    .friendly_name(model="Presence Sensor FP1E", manufacturer="Aqara")
+    .adds(DeviceTemperature)
+    .adds(OccupancySensing)
+    .adds(
+        IasZone,
+        constant_attributes={
+            IasZone.AttributeDefs.zone_type: IasZone.ZoneType.Motion_Sensor
+        },
+    )
```

Comment:
If these clusters are completely virtual/fake (not present on the device), we'd probably want them to be a `LocalDataCluster`. E.g.:

```python
class IasZoneLocal(LocalDataCluster, IasZone):
    """Virtual cluster for IasZone."""

    # required to make sure ZHA creates sensors when initially pairing
    _VALID_ATTRIBUTES = {IasZone.AttributeDefs.zone_status.id}
```

`_VALID_ATTRIBUTES` is probably required to make sure that ZHA creates the sensors when the device is initially paired (so completely removed from ZHA before pairing again), even when there's no initial data in the zigpy DB (which there is not, as the attribute cannot be read on pairing).

Reply:
> It might be a good idea for us to provide a shorthand v2 quirk method for this. But that does't exist yet..

Yes I think that would be a great addition, in the meantime I've updated the quirk to use Local clusters for motion and occupancy.